src/channel.cr

```diff
@@ -111,9 +117,11 @@ abstract class Channel(T)
         return ops.size, nil
       end
 
+      fiber = Fiber.current
+      fiber.callback = ->{ ops.each &.unwait(fiber) }
```

Comment:
This forms a closure and so it allocates memory, which is not ideal. Please change it to store a reference to `ops` directly in `Fiber`. For example you can have `record UnwaitCallback, ops : ...` and then `callback : UnwaitCallback` or something like that (and if we need more callbacks we create a union and son on).

Reply:
@asterite I tried implementing your suggestion but I am not able to figure out the `type` for `ops`. If I use `Tuple | Array`, I get the following error:
```
can't use Array(T) in unions yet, use a more specific type
```

Reply:
Oh, right, it can't be implemented. Nevermind.

Reply:
Would converting the tuple to an array be more efficient than a closure?

Reply:
The problem is that w don't know the types inside the array

Reply:
Yeah, right...
Yet we know it's all `Channel`, we just don't know the generic arguments. But that's actually irrelevant for calling `#unwait`. Just thinking here, but we could hack around it by introducing a module which is included in `Channel`:
```cr
abstract class Channel(T)
  module Unwaitable
    def unwait(fiber : Fiber)
    end
  end
    
  include Unwaitable
end

record UnwaitCallback, ops : Array(Channel::Unwaitable)
```

It works. Question is, is it worth the effort of having an additional type and needing to explicitly cast tuple and array arguments to `Array(Channel::Unwaitable)`? Probably not.